File: ios/SwiftKotlination/SwiftKotlination/Sources/Network/Request.swift
import Foundation

enum Request: Hashable {
    case fetchImage(String)
    case fetchTopStories

    var url: String {
        switch self {
        case .fetchTopStories:
            return "https://api.nytimes.com/svc/topstories/v2/home.json"
        case .fetchImage(let url):
            return url
        }
    }

    var method: HTTPMethod {
        switch self {
        case .fetchTopStories,
             .fetchImage:
            return .get
        }
    }

    var parameters: Parameters {
        switch self {
        case .fetchTopStories:
            return .url(["api-key": "de87f25eb97b4f038d8360e0de22e1dd"])
        case .fetchImage:
            return .none
        }
    }
}

File: android/SwiftKotlination/app/src/main/java/fr/jhandguy/swiftkotlination/network/Request.kt
package fr.jhandguy.swiftkotlination.network

sealed class Request {
    object FetchTopStories: Request()
    data class FetchImage(val imageUrl: String): Request()

    val url: String
        get() = when(this) {
            is FetchTopStories  -> "https://api.nytimes.com/svc/topstories/v2/home.json"
            is FetchImage       -> imageUrl
        }

    val method: HTTPMethod
        get() = when(this) {
            is FetchTopStories  -> HTTPMethod.GET
            is FetchImage       -> HTTPMethod.GET
        }

    val parameters: Parameters
        get() = when(this) {
            is FetchTopStories  -> Parameters.Url(listOf(Pair("api-key", "de87f25eb97b4f038d8360e0de22e1dd")))
            is FetchImage       -> Parameters.None
        }
}